utils.ts

```diff
@@ -0,0 +1,53 @@
+import fetch from "node-fetch";
+import nacl from "tweetnacl";
+import rbjs from "random-bytes-js";
+import { state } from "membrane";
+type Method = "GET" | "POST" | "PUT" | "DELETE" | "PATCH";
+
+export async function api(method: Method, path: string, query?: any, body?: string) {
+  if (!state.token) {
+    throw new Error("You must authenticated to use this API. Visit the program's endpoint");
+  }
+  if (query) {
+    Object.keys(query).forEach((key) => (query[key] === undefined ? delete query[key] : {}));
+  }
+  const querystr = query && Object.keys(query).length ? `?${new URLSearchParams(query)}` : "";
+
+  // Using curl's user-agent because Discord's API doesn't like Membrane's default
+  return await fetch(`https://discord.com/api/${path}${querystr}`, {
```

Comment:
Let's start checking for the status code in `api` so that errors bubble up to the user